### ISSUE TITLE
Prebid v7.39.0: drop in Rubicon delivery

### ISSUE THREAD

**Issue by davidballester** [prebid/Prebid.js#11680]
## Type of issue

A question regarding Rubicon. 

## Description

We're updating Prebid.js from v7.38.0 to v7.39.0. This causes a drop on our Rubicon delivery.

This is a sample payload to the `auction` endpoint we make when using v7.38.0. Please notice I've omitted the consent string:

```
{
  "id": "7fe816ef-4bde-4187-93c6-de0e0e6d50b1",
  "test": 0,
  "cur": [
    "USD"
  ],
  "source": {
    "tid": "7fe816ef-4bde-4187-93c6-de0e0e6d50b1"
  },
  "tmax": 3000,
  "imp": [
    {
      "id": "carodaVirtualSlot1",
      "secure": 1,
      "ext": {
        "rubicon-rubicon": {
          "accountId": 25432,
          "siteId": 512596,
          "zoneId": 3075890,
          "video": {
            "language": "sk",
            "size_id": 203
          },
          "position": "btf"
        },
        "tid": "7fe816ef-4bde-4187-93c6-de0e0e6d50b1",
        "prebid": {
          "auctiontimestamp": 1717494856190
        }
      },
      "video": {
        "api": [
          2
        ],
        "context": "outstream",
        "h": 347,
        "linearity": 1,
        "maxbitrate": 29494983,
        "maxduration": 120,
        "mimes": [
          "video/mp4",
          "video/x-ms-wmv",
          "video/webm",
          "application/javascript",
          "video/ogg"
        ],
        "minbitrate": 104,
        "minduration": 5,
        "placement": 3,
        "playbackmethod": [
          6
        ],
        "playerSize": [
          [
            617,
            347
          ]
        ],
        "plcmt": 4,
        "protocols": [
          2,
          3,
          5,
          6,
          7,
          8
        ],
        "skip": 0,
        "useCacheKey": false,
        "w": 617,
        "pos": 3
      }
    }
  ],
  "ext": {
    "prebid": {
      "channel": {
        "name": "pbjs",
        "version": "v7.38.0"
      },
      "cache": {
        "vastxml": {
          "returnCreative": false
        }
      },
      "targeting": {
        "includewinners": true,
        "includebidderkeys": false,
        "pricegranularity": {
          "ranges": [
            {
              "max": 20,
              "increment": 0.1
            }
          ]
        }
      },
      "bidders": {
        "rubicon": {
          "integration": "pbjs"
        }
      },
      "aliases": {
        "rubicon-rubicon": "rubicon"
      }
    }
  },
  "site": {
    "page": "https://www.cas.sk/clanok/2925975/babis-s-monikou-prekvapuju-najskor-rozchod-a-teraz-foto-hovori-za-vsetko/",
    "content": {
      "language": "sk"
    },
    "ref": "https://www.cas.sk/clanok/2927547/mato-durinda-ako-hokejista-prehovoril-o-drsnych-zraneniach-spomenul-aj-moment-ktory-mu-zmenil-zivot/",
    "domain": "www.cas.sk",
    "publisher": {
      "domain": "cas.sk"
    },
    "mobile": 0,
    "keywords": "hokej,andrej babis,rozvod,zapas,ms,monika babisova",
    "ext": {
      "data": {}
    }
  },
  "device": {
    "language": "en",
    "w": 970,
    "h": 966,
    "dnt": 0,
    "ua": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/125.0.0.0 Safari/537.36",
    "lmt": 0,
    "sua": {
      "source": 2,
      "platform": {
        "brand": "Linux",
        "version": [
          "6",
          "5",
          "0"
        ]
      },
      "browsers": [
        {
          "brand": "Google Chrome",
          "version": [
            "125",
            "0",
            "6422",
            "141"
          ]
        },
        {
          "brand": "Chromium",
          "version": [
            "125",
            "0",
            "6422",
            "141"
          ]
        },
        {
          "brand": "Not.A/Brand",
          "version": [
            "24",
            "0",
            "0",
            "0"
          ]
        }
      ],
      "mobile": 0,
      "model": "",
      "bitness": "64",
      "architecture": "x86"
    }
  },
  "regs": {
    "ext": {
      "gdpr": 1
    }
  },
  "user": {
    "ext": {
      "consent": "<CONSENT_STRING>",
      "data": {}
    }
  }
}
```

When using v7.39.0, this is how the payload looks. Again, I've omitted consents:

```
{
  "imp": [
    {
      "ext": {
        "tid": "159b93c5-1165-40f0-8cfa-2bd4a8c788c5",
        "prebid": {
          "bidder": {
            "rubicon-rubicon": {
              "accountId": 25432,
              "siteId": 512596,
              "zoneId": 3075890,
              "video": {
                "language": "sk",
                "size_id": 203
              },
              "position": "btf"
            }
          },
          "adunitcode": "carodaVirtualSlot1"
        }
      },
      "id": "carodaVirtualSlot1",
      "video": {
        "api": [
          2
        ],
        "h": 347,
        "linearity": 1,
        "maxbitrate": 29494983,
        "maxduration": 120,
        "mimes": [
          "video/mp4",
          "video/x-ms-wmv",
          "video/webm",
          "application/javascript",
          "video/ogg"
        ],
        "minbitrate": 104,
        "minduration": 5,
        "placement": 3,
        "playbackmethod": [
          6
        ],
        "protocols": [
          2,
          3,
          5,
          6,
          7,
          8
        ],
        "skip": 0,
        "w": 617,
        "pos": 3
      }
    }
  ],
  "device": {
    "w": 970,
    "h": 966,
    "dnt": 0,
    "ua": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/125.0.0.0 Safari/537.36",
    "language": "en",
    "lmt": 0,
    "sua": {
      "source": 2,
      "platform": {
        "brand": "Linux",
        "version": [
          "6",
          "5",
          "0"
        ]
      },
      "browsers": [
        {
          "brand": "Google Chrome",
          "version": [
            "125",
            "0",
            "6422",
            "141"
          ]
        },
        {
          "brand": "Chromium",
          "version": [
            "125",
            "0",
            "6422",
            "141"
          ]
        },
        {
          "brand": "Not.A/Brand",
          "version": [
            "24",
            "0",
            "0",
            "0"
          ]
        }
      ],
      "mobile": 0,
      "model": "",
      "bitness": "64",
      "architecture": "x86"
    }
  },
  "site": {
    "page": "https://www.cas.sk/clanok/2925975/babis-s-monikou-prekvapuju-najskor-rozchod-a-teraz-foto-hovori-za-vsetko/",
    "ref": "https://www.cas.sk/clanok/2927547/mato-durinda-ako-hokejista-prehovoril-o-drsnych-zraneniach-spomenul-aj-moment-ktory-mu-zmenil-zivot/",
    "domain": "www.cas.sk",
    "publisher": {
      "domain": "cas.sk"
    },
    "mobile": 0,
    "keywords": "hokej,andrej babis,rozvod,zapas,ms,monika babisova",
    "ext": {
      "data": {}
    }
  },
  "user": {
    "ext": {
      "consent": "<CONSENT_STRING>",
      "ConsentedProvidersSettings": {
        "consented_providers": "<CONSENTED_PROVIDERS>"
      },
      "data": {}
    }
  },
  "regs": {
    "ext": {
      "gdpr": 1
    }
  },
  "id": "707b7cf3-8881-4ccb-9e06-8b969bad0948",
  "test": 0,
  "tmax": 3000,
  "source": {
    "tid": "707b7cf3-8881-4ccb-9e06-8b969bad0948"
  },
  "cur": [
    "USD"
  ],
  "ext": {
    "prebid": {
      "auctiontimestamp": 1717495046639,
      "targeting": {
        "includewinners": true,
        "includebidderkeys": false,
        "pricegranularity": {
          "ranges": [
            {
              "max": 20,
              "increment": 0.1
            }
          ]
        }
      },
      "channel": {
        "name": "pbjs",
        "version": "v7.39.0"
      },
      "aliases": {
        "rubicon-rubicon": "rubicon"
      },
      "cache": {
        "vastxml": {
          "returnCreative": false
        }
      },
      "bidders": {
        "rubicon": {
          "integration": "pbjs"
        }
      }
    }
  }
}
```

I've tried to analyze the changes in `rubiconBidAdapter.js` code from one version to the other. But the changes are massive, so I'm unsure of where the issue could be. 😅

Any help would be really appreciated!

### Expected results

I expected our delivery for Rubicon not to be affected by updating our Prebid.js from v7.38.0 to v7.39.0.

### Actual results

Our delivery drops.

**Comment by dgirardi** [comment]
May be related to https://github.com/prebid/Prebid.js/pull/9605

Some notable differences between old and new versions:

1. new version is missing `plcmt`
2. new version has bidder params under `imp[].ext.prebid.bidder["rubicon-rubicon"]` instead of `imp[].ext["rubicon-rubicon"]`
3. new version is missing `site.content.language`

**Comment by davidballester** [comment]
Hi, @dgirardi, and thanks for your quick response.

Yes, I imagine our issues are triggered by the changes from that PR. 

I noticed, however, that according to the unit tests modified on that same PR, the position of the bidder params is supposed to be correct ([here](https://github.com/prebid/Prebid.js/pull/9605/files#diff-dfdf2a89b2642fcc755d25efea64d07e40adb69ade259519482847381cef69b0R1576-R1579)).

We haven't changed anything in the way we configure the bidder when updating from v7.38.0 to v7.39.0, and I could not find any mention in the docs about changes I should made to video format bids. 

I'll look into the missing `plcmt` and `site.content.language`, but I'm not sure if they are enough to explain a total drop.

**Comment by davidballester** [comment]
For completion, this is the ad unit we use:

```
{
  "bids": [
    {
      "bidder": "rubicon-rubicon",
      "alias": "rubicon-rubicon",
      "params": {
        "accountId": 25432,
        "siteId": 512596,
        "zoneId": 3075890,
        "video": {
          "language": "sk",
          "size_id": 203
        },
        "position": "btf"
      },
      "crumbs": {
        "pubcid": "d86a0c82-cf6d-4486-8448-97777ff7e8a4"
      }
    }
  ],
  "code": "carodaVirtualSlot1",
  "mediaTypes": {
    "video": {
      "api": [
        2
      ],
      "context": "outstream",
      "h": 84,
      "linearity": 1,
      "maxbitrate": 29494983,
      "maxduration": 120,
      "mimes": [
        "video/mp4",
        "video/x-ms-wmv",
        "video/webm",
        "application/javascript",
        "video/ogg"
      ],
      "minbitrate": 104,
      "minduration": 5,
      "placement": 3,
      "playbackmethod": [
        6
      ],
      "playerSize": [
        [
          392,
          220
        ]
      ],
      "plcmt": 4,
      "protocols": [
        2,
        3,
        5,
        6,
        7,
        8
      ],
      "skip": 0,
      "useCacheKey": false,
      "w": 149
    }
  },
  "renderer": {
    "url": "data://data:text/html,"
  }
}
```

We use the same for both v7.38.0 and v7.39.0.

**Comment by robertrmartinez** [comment]
The main change between these two versions was that our VIDEO calls begain using the ORTB Converter Prebid Library instead of building the Payload itself.

So - a lot of changes happened.

But one thing that recently came up that we have seen did potentially effect monetization was solved with this PR https://github.com/prebid/Prebid.js/pull/12108

TL;DR

Device wxh was being incorrectly set by the converter and has been fixed!
